components/match2/wikis/trackmania/match_summary.lua

```diff
@@ -0,0 +1,363 @@
+end
+
+function Header:rightOpponentTeam(content)
+	self.rightElementAdditional = content
+	return self
+end
+
+function Header:scoreBoard(content)
+	self.scoreBoard = content
+	return self
+end
+
+function Header:leftOpponent(content)
+	self.leftElement = content
+	return self
+end
+
+function Header:rightOpponent(content)
+	self.rightElement = content
+	return self
+end
+
+function Header:createScoreDisplay(opponent1, opponent2)
+	local function getScore(opponent)
+		local scoreText
+		local isWinner = opponent.placement == 1 or opponent.advances
+		if opponent.placement2 then
+			-- Bracket Reset, show W/L
+			if opponent.placement2 == 1 then
+				isWinner = true
+				scoreText = 'W'
+			else
+				isWinner = false
+				scoreText = 'L'
+			end
+		elseif opponent.extradata and opponent.extradata.additionalScores then
+			-- Match Series (Sets), show the series score
+			scoreText = (opponent.extradata.set1win and 1 or 0)
+					+ (opponent.extradata.set2win and 1 or 0)
+					+ (opponent.extradata.set3win and 1 or 0)
+		else
+			scoreText = OpponentDisplay.InlineScore(opponent)
+		end
+		return OpponentDisplay.BlockScore{
+			isWinner = isWinner,
+			scoreText = scoreText,
+		}
+	end
+
+	return mw.html.create('div')
+		:addClass('brkts-popup-spaced')
+		:node(
+			getScore(opponent1)
+				:css('margin-right', 0)
+		)
+		:node(' : ')
+		:node(getScore(opponent2))
+end
+
+function Header:createScoreBoard(score, bestof, isNotFinished)
+	local scoreBoardNode = mw.html.create('div')
+		:addClass('brkts-popup-spaced')
+
+	if String.isNotEmpty(bestof) and bestof > 0 and isNotFinished then
+		return scoreBoardNode
+			:node(mw.html.create('span')
+				:css('line-height', '1.1')
+				:css('width', '100%')
+				:css('text-align', 'center')
+				:node(score)
+			)
+			:node('<br>')
+			:node(mw.html.create('span')
+				:wikitext('(')
+				:node(Abbreviation.make(
+					'Bo' .. bestof,
+					'Best of ' .. bestof
+				))
+				:wikitext(')')
+			)
+	end
+
+	return scoreBoardNode:node(score)
+end
+
+function Header:createOpponent(opponent, opponentIndex)
+	return OpponentDisplay.BlockOpponent({
+		flip = opponentIndex == 1,
+		opponent = opponent,
+		overflow = 'ellipsis',
+		teamStyle = 'short',
+	})
+end
+
+function Header:create()
+	self.root:tag('div'):addClass('brkts-popup-header-opponent'):addClass('brkts-popup-header-opponent-left')
+		:node(self.leftElementAdditional)
+		:node(self.leftElement)
+	self.root:node(self.scoreBoard)
+	self.root:tag('div'):addClass('brkts-popup-header-opponent'):addClass('brkts-popup-header-opponent-right')
+		:node(self.rightElement)
+		:node(self.rightElementAdditional)
+	return self.root
+end
+
+
+local CustomMatchSummary = {}
+
+function CustomMatchSummary._getHeadToHead(opponents)
+	local team1, team2 = mw.uri.encode(opponents[1].name), mw.uri.encode(opponents[2].name)
+	local link = tostring(mw.uri.fullUrl('Special:RunQuery/Head2head'))
+		.. '?RunQuery=Run&pfRunQueryFormName=Head2head&Headtohead%5Bteam1%5D='
```

Comment:
imo check #2288 for a more structured way to build the link

Reply:
https://github.com/Liquipedia/Lua-Modules/pull/2312/commits/05610b9cc64246f2f7e38b6d938f4bdd82f28bec